job-server/src/spark.jobserver/cassandra/Cassandra.scala

```diff
@@ -0,0 +1,76 @@
+package spark.jobserver.cassandra
+
+import com.datastax.driver.core.{BoundStatement, ResultSet, ResultSetFuture}
+import scala.concurrent.{CanAwait, Future, ExecutionContext}
+import scala.util.{Success, Try}
+import scala.concurrent.duration.Duration
+import java.util.concurrent.{TimeUnit, Executor => JExecutor}
+
+
+trait CassandraResultSetOperations {
+
+  private case class ExecutionContextExecutor(executionContext: ExecutionContext) extends JExecutor {
+    def execute(command: Runnable): Unit = { executionContext.execute(command) }
+  }
+
+  protected class RichResultSetFuture(resultSetFuture: ResultSetFuture) extends Future[ResultSet] {
+    @throws(classOf[InterruptedException])
+    @throws(classOf[scala.concurrent.TimeoutException])
+    def ready(atMost: Duration)(implicit permit: CanAwait): this.type = {
+      resultSetFuture.get(atMost.toMillis, TimeUnit.MILLISECONDS)
+      this
+    }
+
+    @throws(classOf[Exception])
+    def result(atMost: Duration)(implicit permit: CanAwait): ResultSet = {
+      resultSetFuture.get(atMost.toMillis, TimeUnit.MILLISECONDS)
+    }
+
+    def onComplete[U](func: (Try[ResultSet]) => U)(implicit executionContext: ExecutionContext): Unit = {
+      if (resultSetFuture.isDone) {
+        func(Success(resultSetFuture.getUninterruptibly))
+      } else {
+        resultSetFuture.addListener(new Runnable {
+          def run() {
+            func(Try(resultSetFuture.get()))
+          }
+        }, ExecutionContextExecutor(executionContext))
+      }
+    }
+
+    def isCompleted: Boolean = resultSetFuture.isDone
+
+    def value: Option[Try[ResultSet]] = if (resultSetFuture.isDone) Some(Try(resultSetFuture.get())) else None
+  }
+
+  implicit def toFuture(resultSetFuture: ResultSetFuture): Future[ResultSet] = {
+    new RichResultSetFuture(resultSetFuture)
+  }
+}
+
+trait Binder[-A] {
+
```

Comment:
@velvia some style guides require you to have a blank line in those places and at the end of the file.